Add support for multiple consoles/windows.

fkallevik opened the pull request:
Hey! Here is a suggestion for how a solution to issue #5 could look like.

Any feedback will be appreciated :)

RobLoach commented:
That's very cool! Was able to run the SDL demo and switch between the windows. Couple questions...

1. I see the change required to the demo code. Would it be possible to have the renderer demos keep the same code?
2. There's a `nk_console_init_multiple()` that registeres a number of different `nk_console`. Would it be possible to keep the single parent `nk_console`, and then have a number of different `nk_console_window`s added to it? Or is that backwards and just wouldn't work?

fkallevik commented:
I refactored the solution and added the `NK_CONSOLE_WINDOW` widget type. Let me know what you think about this way of doing it, whenever you get the chance to look over it :)

RobLoach commented:
Was playing around with this yesterday. It's really cool. Been thinking about that demo set up, and considering making it a bit easier to grok as an example for how to use the multiple windows.

Is there anything you think should be in before we merge this?

fkallevik commented:
Cool, I'm glad that you like it!

At the very least I should make sure that all the other demos work, at the moment they are broken because of the changes to `demo/common/nuklear_console_demo.c`.

Should we make two demos for each rendering lib as well? One for single window and one for multiple windows?

RobLoach commented:
Been thinking about the design a bit, and put together a small thought experiment over at https://github.com/fkallevik/nuklear_console/pull/1 .

What are your thoughts about the following...
- When you create a nk_console, it automatically makes the first window for you
- When adding a widget to an nk_console, if you're adding it to the top, it defaults to adding it to the first window found
- This would allow you to create and destruct the windows as your wish. Good for little pop ups.
- Rather than leveraging window index and window_count, use cvector
- We can redesign all the usage of nk_console_render_window() and such, move all the data into nk_console_window_data. Don't worry about backwards compatibility for this change

Have been thinking of the optimal way to use this, and I think having everything move to the windowing method is likely best.

fkallevik commented:
Great food for thought!

* Automatically creating the first window would be nice, but how do we go about determining the window title, size, position and flags?
* I like the idea of adding the "Multiple windows" button to the original demo, keeping it simple. Maybe it could register a new window popup on-the-fly when clicked, and then a button in the new window could destruct it again.
* Yes, `window_count` is uneccessary, can just use `cvector_size`.
* How do you suggest we track which window is active, so TAB/LT/RT can be used to switch focus between them, if not the `active_window_index`?
* Do you envision the user calling `nk_begin` and `nk_end` like before? When windows are a part of `nk_console` it almost doesn't make sense anymore, and would maybe be cleaner if they can simply call `nk_console_render(console)` in their update loop?
